crates/router/src/core/errors.rs

```diff
@@ -80,6 +81,67 @@ pub enum StorageError {
     RedisError(error_stack::Report<RedisError>),
 }
 
+impl ErrorSwitch<DataStorageError> for StorageError {
+    fn switch(&self) -> DataStorageError {
+        self.into()
+    }
+}
+
+#[allow(clippy::from_over_into)]
+impl Into<DataStorageError> for &StorageError {
+    fn into(self) -> DataStorageError {
+        match self {
+            StorageError::DatabaseError(i) => match i.current_context() {
+                storage_errors::DatabaseError::DatabaseConnectionError => {
+                    DataStorageError::DatabaseConnectionError
+                }
+                // TODO: Update this error type to encompass & propagate the missing type (instead of generic `db value not found`)
+                storage_errors::DatabaseError::NotFound => {
+                    DataStorageError::ValueNotFound(String::from("db value not found"))
+                }
+                // TODO: Update this error type to encompass & propagate the duplicate type (instead of generic `db value not found`)
+                storage_errors::DatabaseError::UniqueViolation => {
+                    DataStorageError::DuplicateValue {
+                        entity: "db entity",
+                        key: None,
+                    }
+                }
```

Comment:
Would this be done now, or as part of a later change?

Reply:
later since this involves changes to the more commonly used Database Error affecting all storage trait implementations